js/ui/map.js

```diff
@@ -855,41 +856,46 @@ util.extend(Map.prototype, /** @lends Map.prototype */{
      * @private
      */
     _render: function() {
-        if (this.style && this._styleDirty) {
-            this._styleDirty = false;
-            this.style.update(this._classes, this._classOptions);
-            this._classOptions = null;
-            this.style._recalculate(this.transform.zoom);
-        }
-
-        if (this.style && this._sourcesDirty) {
-            this._sourcesDirty = false;
-            this.style._updateSources(this.transform);
-        }
-
-        this.painter.render(this.style, {
-            debug: this.showTileBoundaries,
-            showOverdrawInspector: this._showOverdrawInspector,
-            vertices: this.vertices,
-            rotating: this.rotating,
-            zooming: this.zooming
-        });
-
-        this.fire('render');
-
-        if (this.loaded() && !this._loaded) {
-            this._loaded = true;
-            this.fire('load');
-        }
-
-        this._frameId = null;
-
-        if (!this.animationLoop.stopped()) {
-            this._styleDirty = true;
-        }
-
-        if (this._sourcesDirty || this._repaint || !this.animationLoop.stopped()) {
-            this._rerender();
+        try {
+            if (this.style && this._styleDirty) {
+                this._styleDirty = false;
+                this.style.update(this._classes, this._classOptions);
+                this._classOptions = null;
+                this.style._recalculate(this.transform.zoom);
+            }
+
+            if (this.style && this._sourcesDirty) {
+                this._sourcesDirty = false;
+                this.style._updateSources(this.transform);
+            }
+
+            this.painter.render(this.style, {
+                debug: this.showTileBoundaries,
+                showOverdrawInspector: this._showOverdrawInspector,
+                vertices: this.vertices,
+                rotating: this.rotating,
+                zooming: this.zooming
+            });
+
+            this.fire('render');
+
+            if (this.loaded() && !this._loaded) {
+                this._loaded = true;
+                this.fire('load');
+            }
+
+            this._frameId = null;
+
+            if (!this.animationLoop.stopped()) {
+                this._styleDirty = true;
+            }
+
+            if (this._sourcesDirty || this._repaint || !this.animationLoop.stopped()) {
+                this._rerender();
+            }
+
+        } catch (error) {
+            this.fire('error', error);
```

Comment:
Hm. On further thought, I don't think it makes so much sense to have both an error event and a thrown error. The render loop will still end because the `this._rerender` call is in the `try` block.